### PR TITLE
DTS:26932 Lead Time Enhc

### PR DESCRIPTION
**Pull request by shunaray**

### CODE REVIEW

**Review by hirbabar — DISMISSED** [review]

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImpl.java [hunk 1]
```diff
@@ -150,8 +151,9 @@ public KpiElement getKpiData(KpiRequest kpiRequest, KpiElement kpiElement,
 	@SuppressWarnings("unchecked")
 	private void projectWiseLeafNodeValue(List<Node> leafNodeList, KpiElement kpiElement, KpiRequest kpiRequest,
 			DataCount dataCount) {
-
-		Map<String, Object> resultMap = fetchKPIDataFromDb(leafNodeList, null, null, kpiRequest);
+		String startDate = "2023-07-04";
+		String endDate = "2023-07-07";
```
Comment (aksshriv1) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImpl.java:155]: Why hard coded date is maintained?

File: customapi/src/test/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImplTest.java [hunk 2]
```diff
@@ -177,7 +177,7 @@ public void testFetchKPIDataFromDBData() throws ApplicationException {
 		List<Node> leafNodeList = new ArrayList<>();
 		leafNodeList = KPIHelperUtil.getLeafNodes(treeAggregatorDetail.getRoot(), leafNodeList);
 		when(configHelperService.getFieldMappingMap()).thenReturn(fieldMappingMap);
-		when(jiraIssueCustomHistoryRepository.findByFilterAndFromStatusMap(any(), any()))
+		when(jiraIssueCustomHistoryRepository.findByFilterAndFromStatusMapWithDateFilter(any(), any(),any(), any()))
```
Comment (aksshriv1) [inline customapi/src/test/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImplTest.java:180]: check formatting here.

File: customapi/src/test/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImplTest.java [hunk 3]
```diff
@@ -192,7 +192,7 @@ public void testGetKpiData() throws ApplicationException {
 		TreeAggregatorDetail treeAggregatorDetail = KPIHelperUtil.getTreeLeafNodesGroupedByFilter(kpiRequest,
 				accountHierarchyDataList, new ArrayList<>(), "hierarchyLevelOne", 5);
 		when(configHelperService.getFieldMappingMap()).thenReturn(fieldMappingMap);
-		when(jiraIssueCustomHistoryRepository.findByFilterAndFromStatusMap(any(), any()))
+		when(jiraIssueCustomHistoryRepository.findByFilterAndFromStatusMapWithDateFilter(any(), any(),any(), any()))
```
Comment (aksshriv1) [inline customapi/src/test/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImplTest.java:195]: check formatting here

File: customapi/src/test/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImplTest.java (hunk 2 above)
Comment (shunaray) [reply customapi/src/test/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImplTest.java]: done

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImpl.java (hunk 1 above)
Comment (shunaray) [reply customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImpl.java]: start & endDate will come in kpiReq from UI now updated the code

**Review by hirbabar — APPROVED** [review]

**Review by aksshriv1 — APPROVED** [review]

**Pull request merged**